### ISSUE TITLE
Question: why does hlint complaint about `Redundant lambda`?

### ISSUE THREAD

**Issue by pjljvandelaar** [ndmitchell/hlint#467]
hlint complaints about `redundant lambda`.
Yet on https://downloads.haskell.org/~ghc/7.0.3/docs/html/users_guide/pragmas.html
it is stated:
```
Moreover, GHC will only inline the function if it is fully applied, where "fully applied" means applied to as many arguments as appear (syntactically) on the LHS of the function definition. For example: 
comp1 :: (b -> c) -> (a -> b) -> a -> c
{-# INLINE comp1 #-}
comp1 f g = \x -> f (g x)

comp2 :: (b -> c) -> (a -> b) -> a -> c
{-# INLINE comp2 #-}
comp2 f g x = f (g x)


The two functions comp1 and comp2 have the same semantics, but comp1 will be inlined when applied to two arguments, while comp2 requires three. This might make a big difference if you say 
map (not `comp1` not) xs


which will optimise better than the corresponding use of `comp2`. 
```
So there is a good reason to use lambda expressions in definitions and I had not expect this warning.
But probably I miss something...

**Comment by ndmitchell** [comment]
HLint has the goal of making Haskell code cleaner - not faster. Some of the changes may make it faster. Some of the changes may inhibit GHC's optimiser to a small extent. If you've hyper-optimised your code to the extent you think it will matter, you shouldn't even be breathing on it. In general it probably makes no difference to the optimiser, but if the code is clearer, humans will be grateful.
